core/core.php

```diff
@@ -181,4 +185,93 @@ public function validate_badge_order($user_id)
 
 		return true;
 	}
+
+	/**
+	 * Gets the url to the user avatar.
+	 * Returns empty string on failure or if there is no avatar.
+	 *
+	 * @param array $user_data Avatar data from the user table
+	 * @return array
+	 */
+	public function get_avatar_data($user_data)
+	{
+		if (!$this->config['allow_avatar'])
+		{
+			return false;
+		}
+
+		$row = [
+			'avatar' => $user_data['user_avatar'],
+			'avatar_width' => $user_data['user_avatar_width'],
+			'avatar_height' => $user_data['user_avatar_height'],
+		];
+
+		$avatar_data = [
+			'src' => $row['avatar'],
+			'width' => $row['avatar_width'],
+			'height' => $row['avatar_height'],
+		];
+
+		$driver = $this->avatar_manager->get_driver($user_data['user_avatar_type']);
+
+		if (!$driver)
+		{
+			return false;
+		}
+
+		$avatar_data = $driver->get_data($row);
+
+		if ($user_data['user_avatar_type'] === 'avatar.driver.gravatar')
+		{
+			$avatar_data['src'] = $this->get_gravatar_url($row);
+		} else if ($user_data['user_avatar_type'] === 'avatar.driver.upload') {
+			$avatar_data['src'] = $this->get_upload_avatar_url($user_data['user_avatar']);
+		}
+
+		return $avatar_data;
+	}
+
+	/**
+	 * Gets the full URL for a user uploaded avatar.
+	 * 
+	 * @param $user_avatar User avatar data
+	 * @return string Avatar URL
+	 */
+	protected function get_upload_avatar_url($user_avatar)
+	{
+		return generate_board_url() . '/download/file.php' . '?avatar=' . $user_avatar;
+	}
```

Comment:
I had to add this function because otherwise phpBB gives a relative URL for the avatar.

Reply:
Do we need a similar special case for gallery avatars?

Reply:
Good catch - I added a method to handle that case.